src/generators/mappersGenerator.ts

```diff
@@ -21,29 +28,132 @@ export function generateMappers(
     moduleSpecifier: "@azure/core-http"
   });
 
-  for (const mapper of clientDetails.mappers) {
-    // Generate the mapper definition and replace object key strings
-    // with the unquoted name
-    const mapperString = JSON.stringify(mapper, undefined, 2).replace(
-      /\"([^"]+)\":/g,
-      "$1:"
-    );
+  let discriminators: { [key: string]: string } = {};
 
+  for (const mapper of clientDetails.mappers) {
     if (mapper.type.name === MapperType.Composite && mapper.serializedName) {
+      const compositeMapper = mapper as CompositeMapper;
+
+      if (compositeMapper.type.polymorphicDiscriminator) {
+        const { uberParent, className } = compositeMapper.type;
+        if (!uberParent || !className) {
+          throw new Error(
+            `Expected CompositeMapper with polymorphicDiscriminator to specify uberParent property uberParent: ${uberParent} className: ${className}`
+          );
+        }
+
+        if (uberParent === className) {
+          discriminators[className] = className;
```

Comment:
Should `serializedName` be used as the key here, or only when the `uberParent` is used?

Reply:
Yes good catch it should always be serializedName